deploy/docker/production/assume_identity.py

```diff
@@ -0,0 +1,43 @@
+# This code snippet is adapated from AWS ECS Documentation
+# The plan is 'create' environments by uploading Secrets to AWS Secrets Manager
+# and retrieve the secrets upon deployment.
+# The idea is the container runner will have an assumed IAM role granting access
+# specifically to the secret
+
+import os
+import logging
+from dcicutils.qa_utils import override_environ
+from dcicutils.deployment_utils import IniFileManager
+from dcicutils.secrets_utils import assume_identity
+
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__file__)
+_MY_DIR = os.path.dirname(__file__)
+
+
+class CGAPDockerIniFileManager(IniFileManager):
+    """ This runs at top level, so path is slightly different. """
+    # should work but doesn't (missing cgap-portal): os.path.join(os.path.dirname(_MY_DIR), "pyproject.toml")
+    # TODO: repair via
+    # expected = <hardwired>
+    # actual = <computed>
+    # assert actual == expected, "The actual value %s was not what we expected, %s." % (actual, expected)
+    TEMPLATE_DIR = '/home/nginx/cgap-portal/deploy/ini_files'
+    PYPROJECT_FILE_NAME = '/home/nginx/cgap-portal/pyproject.toml'
+
+
+def build_production_ini_from_global_application_configuration():
+    """ This function makes a request to secrets manager for the identity passed to the container.
+        See documentation on API in dcicutils.
+    """
+    identity = assume_identity()
+
+    # build production.ini
+    with override_environ(**identity):
+
+        CGAPDockerIniFileManager.main()
```

Comment:
Maybe there should be a subclass of `IniFileManager` in `dcicutils` that knows about the `secrets_utils` and bundles all of this together so that you can just do (hypothesized):
```suggestion
def build_production_ini_from_global_application_configuration():
    """ This function makes a request to secrets manager for the identity passed to the container.
        See documentation on API in dcicutils.
    """
   CGAPDockerIniFileManager.main(assume_identity=True)
```